logstreams/src/test/java/io/zeebe/logstreams/processor/AsyncSnapshotingTest.java

```diff
@@ -178,72 +151,211 @@ public void shouldNotStopTakingSnapshotsAfterFailingReplication() throws Excepti
     // given
     logStreamRule.getClock().addTime(Duration.ofMinutes(1));
-    verify(snapshotController, timeout(TIMEOUT).times(1)).takeTempSnapshot();
+    verify(snapshotController, TIMEOUT.times(1)).takeTempSnapshot();
     logStreamRule.setCommitPosition(99L);
-    verify(snapshotController, timeout(TIMEOUT).times(1)).moveValidSnapshot(25);
+    verify(snapshotController, TIMEOUT.times(1)).moveValidSnapshot(25);
 
     // when
     logStreamRule.getClock().addTime(Duration.ofMinutes(1));
-    verify(snapshotController, timeout(TIMEOUT).times(1)).takeTempSnapshot();
+    verify(snapshotController, TIMEOUT.times(1)).takeTempSnapshot();
     logStreamRule.setCommitPosition(100L);
 
     // then
-    verify(snapshotController, timeout(500).times(1)).moveValidSnapshot(32);
+    verify(snapshotController, TIMEOUT.times(1)).moveValidSnapshot(32);
   }
 
   @Test
   public void shouldInvokeDataDeleteCallbackOnMaxSnapshots() throws IOException {
     // when
     logStreamRule.getClock().addTime(Duration.ofMinutes(1));
-    verify(snapshotController, timeout(500).times(1)).takeTempSnapshot();
+    verify(snapshotController, TIMEOUT.times(1)).takeTempSnapshot();
     logStreamRule.setCommitPosition(99L);
-    verify(snapshotController, timeout(500).times(1)).moveValidSnapshot(25);
+    verify(snapshotController, TIMEOUT.times(1)).moveValidSnapshot(25);
 
     logStreamRule.getClock().addTime(Duration.ofMinutes(1));
-    verify(snapshotController, timeout(500).times(1)).takeTempSnapshot();
+    verify(snapshotController, TIMEOUT.times(1)).takeTempSnapshot();
     logStreamRule.setCommitPosition(100L);
-    verify(snapshotController, timeout(500).times(1)).moveValidSnapshot(32);
+    verify(snapshotController, TIMEOUT.times(1)).moveValidSnapshot(32);
+
+    // then
+    verify(mockDeleteCallback, TIMEOUT).noop(eq(32L));
+  }
+
+  @Test
+  public void shouldNotTakeSameSnapshotTwice() throws Exception {
+    // given
+    final long lastProcessedPosition = 25L;
+    final long lastWrittenPosition = 26L;
+    final long commitPosition = 100L;
+
+    when(lastProcessedEventPositionSupplier.get())
+        .thenReturn(CompletableActorFuture.completed(lastProcessedPosition));
+    when(lastWrittenEventPositionSupplier.get())
+        .thenReturn(CompletableActorFuture.completed(lastWrittenPosition));
+    logStreamRule.setCommitPosition(commitPosition);
+
+    logStreamRule.getClock().addTime(Duration.ofMinutes(1));
+    verify(snapshotController, TIMEOUT.times(1)).moveValidSnapshot(lastProcessedPosition);
+    verify(lastProcessedEventPositionSupplier, TIMEOUT).get();
+
+    // when
+    logStreamRule.getClock().addTime(Duration.ofMinutes(1));
+    verify(lastProcessedEventPositionSupplier, TIMEOUT.times(2)).get();
 
     // then
-    verify(mockDeleteCallback, timeout(500)).noop(eq(32L));
+    final InOrder inOrder = Mockito.inOrder(snapshotController);
+    inOrder.verify(snapshotController, TIMEOUT.times(1)).takeTempSnapshot();
```

Comment:
I think it would make sense to have these assertions between the two snapshots and not after since it would also prove that these calls weren't made by the second snapshot attempt. If we left the ```verifyNoMoreInteractions``` after the  ```addTime``` it would also tell us that the snapshotController was't used after the first snapshot.